lib/bolt/inventory/group2.rb

```diff
@@ -162,7 +162,8 @@ def data_merge(data1, data2)
           'config' => Bolt::Util.deep_merge(data1['config'], data2['config']),
           'name' => data1['name'] || data2['name'],
           'uri' => data1['uri'] || data2['uri'],
-          'alias' => data1['alias'] || data2['alias'],
+          # Collect all aliases across all groups for each target uri
+          'alias' => [*data1['alias'], *data2['alias']].sort,
```

Comment:
I think we can use the set notation for union here for arrays `'alias' => data1['alias'] | data2['alias']`. Why do you sort? Is that just for testing? The aliases are guaranteed to be unique which is good, but I think this edge case is very strange. But i guess it makes sense, specifically if you re-create a target with target.new you destroy the target information and re-create. It seems that the only thing that is preserved is the alias. I think this is OK but it is a weird use case. Also luckily it appears to be the same behavior in a plan. 
```puppet
plan test(){
  Target.new('name' => 'foo', 'target_alias' => ['printed', 'boop'])
  out::message("${get_target('boop')}")
  out::message("${get_target('printed')}")
  Target.new('name' => 'foo', 'target_alias' => 'also_printed')
  out::message("${get_target('boop')}")
  out::message("${get_target('printed')}")
  out::message("${get_target('also_printed')}")
}
```
```
$ bolt plan run test
Starting: plan test
Starting: plan test
foo
foo
foo
foo
foo
Finished: plan test in 0.01 sec
Finished: plan test in 0.01 sec
```

Reply:
I don't think the set notation for union alone will work here, since `data1['alias']` and `data2['alias']` could be `nil`. We would need to do something like `data1['alias'].to_a | data2['alias'].to_a` for it to work -- not sure if this or the splat makes more sense.

The sort was just personal preference to have the aliases easy to read through, but I'm realizing we don't do that anywhere else, so it probably doesn't make sense.

Reply:
Oh, neat. Looks like we need to make sure both values are arrays before doing the set union for it to work, since either value can be `nil`.

I added the sort just out of a personal preference to have things ordered nicely... but I'm realizing we dont' do that anywhere else, so I'll remove it.

Reply:
I'm starting to think this is a mistake. Perhaps with `Target.new`, we should clean up old aliases in inventory v2.

Reply:
I think the technically correct thing to do would be to handle it like for example `config`. In the case:
```yaml
version: 2
groups:
  - name: foo 
    targets:
      - name: targ
        alias:
          - printed
          - boop
        config:
          transport: local
  - name: bar 
    targets:
      - name: targ
        alias: [also_printed]
        config:
          transport: winrm
```
I would expect alias to be [printed, boop] and config to be transport local.

Reply:
Updated to remove all existing alia when using `Target.new`.